Platform/Authentication/Result/AuthenticationResult.swift

```diff
@@ -0,0 +1,18 @@
+import Foundation
+
+/// Enumeration that represents the result of the authentication process.
+public enum AuthenticationResult {
```

Comment:
Nitpick on the naming here.   This name implies it should be used as the result of call to `autheticate` when in-fact its the result used for `fetchToken`. 

But we could eliminate this question altogether by just using a `Result` object?

Reply:
The problem I see with mocking `Result` type from Swift 5 in the public API of our SDK is that it will make this workaround publicly available in client's code which might be an unintended side effect.

Reply:
Wouldn't the `Result` effectively be namespaced `PusherPlatform.Result`?

Reply:
I am not afraid of clashing symbols. You are right that it would be in `PusherPlatform` namespace. The problem I am pointing to is that the name `Result` is very generic and the practice of implementing `Result` in Swift 4 is quite popular. All of that makes working with Xcode's code completion more confusing/annoying. Are we ok with that happening to developers using our SDK?

Reply:
I have just tried it to consume a custom `Result` and the compiler is getting rather confused.  So I think you are right, we should not try to define a `Result`.  👍 

Are you happy with the name `AuthenticationResult` vs `FetchTokenResult` though?

Reply:
I have no personal preference here.

Reply:
I'm not sure I have a strong preference either.  When I saw `AuthenticationResult` used with the `fetchToken` func though I started asking myself whether I there was something I had not yet understood yet, thats all.